eng/Testing.targets

```diff
@@ -0,0 +1,15 @@
+<Project>
+  <PropertyGroup Condition=" '$(SkipTests)' == '' and '$(IsTestProject)' == 'true' ">
+    <!-- Skip tests by default unless explicitly set to false -->
+    <SkipTests>true</SkipTests>
+
+    <!-- Only run tests if the build is running on GitHub Actions  -->
+    <SkipTests Condition=" '$(IsGitHubActions)' == 'true' and '$(RunTestsOnGithubActions)' == 'true' ">false</SkipTests>
+
+    <!-- Only run tests if the build is running on Helix infra  -->
+    <SkipTests Condition=" '$(IsHelix)' == 'true' and '$(RunTestsOnHelix)' == 'true' ">false</SkipTests>
+
+    <!-- Otherwise, run tests on AzDO CI agents -->
+    <SkipTests Condition=" '$(RunTestsOnGithubActions)' != 'true' and '$(RunTestsOnHelix)' != 'true' ">true</SkipTests>
```

Comment:
Not sure I'm following when will this be needed. Looks from above that we are always automatically setting to true and only switching to false in the above two cases, so when this runs it seems like if the condition is true then SkipTests would already be true. Am I missing something?

Reply:
Good catch, the value must be inverted. Thank you